### ISSUE TITLE
Versification data

### ISSUE THREAD

**Issue by curiousdannii** [openscriptures/BibleOrgSys#20]
Hi, I made [a project which handles versification systems for OSIS references](https://github.com/curiousdannii/reversify), and as I couldn't find any good source of versification differences, I started [my own data file](https://github.com/curiousdannii/reversify/blob/master/src/transformations.data).

This project has a more systems than mine, but not all of them from what I could tell, so I was wondering if you'd like to work together.

I noticed that in one of the Python files you say "We still lack a REFERENCE Bible versification system with back-and-forth mappings. This is a MAJOR outstanding deficiency". I think the mappings is the important data, which gets lost in some of the other versification data sets on the net as they just list how many verses each chapter has. An annotated source file which could then generate the BibleVersificationSystem_ files would be ideal IMO.

**Comment by RobH123** [comment]
Yes, these files have lists of chapters with max verse numbers, and although none of the examples contain this, the system is built to also handled omitted, combined (often called bridged), and reordered verses. (Perhaps I forgot about split verses???) I hope to use the original Hebrew/Greek versifications as my reference.

However, on pondering over this more in recent years, I've realised that there's really few standards for translations into the 7000+ languages of the world. So possible there's no such thing as a "versification system" (only individual versification instances)??? (Still pondering.)

**Comment by curiousdannii** [comment]
Hmm, well I don't think that most translators would intend to create a new versification system - having so many just confuses people, it's not a good thing! And for example, I think there is only one verse difference between the ESV and the NLT.

Translations definitely would inherit versification systems too. Obviously most are inherited in general from Stephen Langton's original system. What would be ideal for software would be to actually model the inheritance tree. So for example, in my data you can see that the NAB and NJPS have lots of OT changes, but they're the same except for a handful of chapters. They probably inherit from a shared source, maybe the Vulgate? It would be possible to notate the changes from the Vulgate to the base, and then from the NAB to the Vulgate and the NJPS to the Vulgate, which would end up being more concise. If there were several translations that seemed to share some mappings but we couldn't identify what they inherited from, we could make them inherit from an made-up internal system.

And yeah, split verses are definitely needed, because they can't all be modelled as combined verses. For example in Acts 10:48 and 19:40, where compared to the ESV the NAB joins in one and splits in the other. No matter which version you consider the base, you have a split and a join.

**Comment by Arithmeticus** [comment]
FWIW, like @curiousdannii I too have created a file that tries to reconcile versification systems:
https://github.com/Arithmeticus/TAN-bible/blob/master/TAN-A-div/tanakh.TAN-A-div.xml

In this case, I'm not concerned with translations, but critical editions. It's in TAN-A-div format, which, in part, allows users to reconcile segmentation systems of different TAN versions of the same work. The file isn't operable outside the [TAN framework](http://textalign.net), but it should be human readable enough that anyone can copy, paste, and edit for their own purposes.

With @RobH123 I agree there's no single segmentation system. I wrote [an article on this topic](http://www.digitalhumanities.org/dhq/vol/8/2/000181/000181.html) a few years ago, with particular reference to the Old Testament.

**Comment by RobH123** [comment]
See thread starting here: http://www.crosswire.org/pipermail/sword-devel/2018-February/045483.html
